src/foam/nanos/auth/HumanNameTrait.js

```diff
@@ -32,13 +32,17 @@ foam.INTERFACE({
       documentation: 'Full legal name of user. Appends first, middle & last name.',
       transient: true,
       expression: function(firstName, middleName, lastName) {
-        return middleName != '' ? firstName + ' ' + middleName + ' ' + lastName : firstName + ' ' + lastName;
+        return [firstName, middleName, lastName].filter(name => name).join(' ');
       },
       javaGetter: `
-        return ! getMiddleName().equals("")
-          ? getFirstName() + " " + getMiddleName() + " " + getLastName()
-          : getFirstName() + " " + getLastName();
-      `,
+        java.util.List<String> filteredNameList = java.util.Arrays
+          .asList(this.getFirstName(), this.getMiddleName(), this.getLastName())
+          .stream()
+          .filter(name -> !name.isEmpty())
+          .collect(java.util.stream.Collectors.toList());
+            
+        return String.join(" ", filteredNameList);
+      `
```

Comment:
had to use fullpath for the java libraries in the getter because javaImports didn't work for the getter in interface